dd-sdk-android-core/src/main/kotlin/com/datadog/android/v2/core/DatadogCore.kt

```diff
@@ -281,7 +284,11 @@ internal class DatadogCore(
         }
 
         // always initialize Core Features first
-        coreFeature = CoreFeature(internalLogger)
+        coreFeature = if (persistenceExecutorServiceFactory != null) {
+            CoreFeature(internalLogger, persistenceExecutorServiceFactory)
```

Comment:
why not just have a default value instead and make it mandatory in the `CoreFeature` constructor ?

Reply:
If I pull the default value to `DatadogCore`, then I'm leaking the implementation details with some properties coming from `CoreFeature`.